CHANGELOG.md

```diff
@@ -2,7 +2,7 @@
 
 Canonical reference for changes, improvements, and bugfixes for Boundary.
 
-## Next
+## 0.12.2 (2023/04/04)
```

Comment:
Fly by comment: I believe this should actually be `0.12.1` and we should add a new item for `0.12.2` and include a similar changelog entry as https://github.com/hashicorp/boundary/blob/main/CHANGELOG.md#security